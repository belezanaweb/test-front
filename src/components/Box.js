import React, { Fragment } from 'react';
import styled from 'styled-components';

const Title = styled.h4`
  color: var(--color-dark-gray-1);
  text-transform: uppercase;
  margin-left: 1rem;
`

const BoxContent = styled.div`
  background-color: #fff;
  box-shadow: var(--box-shadow-1)
`

const Box = ({
  title,
  children
}) => (
  <Fragment>
    <Title>{title}</Title>
    <BoxContent>{children}</BoxContent>
  </Fragment>
)

export default Box;
