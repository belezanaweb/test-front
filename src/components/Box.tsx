import React from 'react'

import { styled } from '@/style/stitches.config'
import Title from './Title'

const SpacingStyled = styled('div', {
  marginY: '10px'
})

const BoxStyled = styled('div', {
  backgroundColor: 'white',
  marginX: '10px',
  marginY: '5px',
  paddingY: '5px',
  paddingX: '15px',
  boxShadow: '1px 1px 5px $defaultShadow',
  borderRadius: '3px'
})

interface OwnProps {
  title: string
  children: JSX.Element | JSX.Element[]
}

const Box = (props: OwnProps): React.ReactElement => {
  const { title, children } = props

  return (
    <SpacingStyled>
      <Title>{title}</Title>
      <BoxStyled>{children}</BoxStyled>
    </SpacingStyled>
  )
}

export default Box
