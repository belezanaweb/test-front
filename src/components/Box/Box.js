import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

const StyledBox = styled.div`
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  border-radius: 3px;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0, 0, 29, 0.22);
  padding: 12px 12px 0 13px;
  margin: 10px 10px 19px 9px;
`

const StyledBoxTitle = styled.h2`
  color: #999;
  font-size: 14px;
  font-weight: 700;
  line-height: 17px;
  text-transform: uppercase;
  margin-left: 20px;
  margin-bottom: -7px;
`

function Box({ title, children }) {
  return (
    <>
      <StyledBoxTitle>{title}</StyledBoxTitle>
      <StyledBox>{children}</StyledBox>
    </>
  )
}

Box.propTypes = {
  title: PropTypes.string
}

export default Box
