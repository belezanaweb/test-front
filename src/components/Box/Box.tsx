import styled, { CSSObject } from 'styled-components';

interface FlexProps {
  children: React.ReactNode,
  css?: CSSObject,
  direction?: string,
  inline?: boolean,
}

export const Flex = styled.div<FlexProps>`
  display: ${(props) => (props.inline ? 'inline-flex' : 'flex')};
  flex: 1;
  flex-direction: ${(props) => props.direction};
  width: 100%;

  ${({ css }) => css};
`;

Flex.defaultProps = {
  css: {},
  direction: 'row',
  inline: false,
};

interface GridProps {
  children: React.ReactNode;
  css?: CSSObject;
  gap?: string;
  template?: string;
}

export const Grid = styled.div<GridProps>`
  display: grid;
  grid-template-columns: ${({ template }) => template};
  grid-gap: ${({ gap }) => gap};
  width: 100%;

  ${({ css }) => css};
`;

Grid.defaultProps = {
  css: {},
  gap: '20px',
  template: '1fr 1fr',
};

interface BoxProps {
  children: React.ReactNode,
  css?: CSSObject,
}

const Box = styled.div<BoxProps>`
  /* box-shadow: 0 0 0 1px pink inset; */
  display: flex;
  flex: 1;
  flex-direction: row;

  ${({ css }) => css};
`;

Box.defaultProps = {
  css: {},
};

export default Box;
