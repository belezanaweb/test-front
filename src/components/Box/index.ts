export { default, Flex, Grid } from './Box';
