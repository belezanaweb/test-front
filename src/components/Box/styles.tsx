import styled from 'styled-components';

export const Container = styled.div`
  border-radius: 3px;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0, 0, 29, 0.22);
  padding: 12px;
  width: 100%;
`;
