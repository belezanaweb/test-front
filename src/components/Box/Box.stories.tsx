import React from 'react';

import theme, { Theming } from '../../theme';

import Box from './Box';

const Provider = (storyFn: () => React.ReactNode) => (
  <Theming theme={theme}>
    {storyFn()}
  </Theming>
);

export default {
  component: Box,
  decorators: [Provider],
  title: 'Box',
};

export const Custom = () => (
  <Box css={{ display: 'block', padding: '20px' }}>
    <Box css={{
      border: '1px dashed red',
      borderRadius: '50%',
      height: '80px',
      width: '80px',
    }}>&nbsp;</Box>
  </Box>
);
