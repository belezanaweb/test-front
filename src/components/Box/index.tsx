import React from 'react';
import { Container } from './styles';

const Box: React.FC = ({ children }) => {
  return <Container>{children}</Container>;
};

export default Box;
