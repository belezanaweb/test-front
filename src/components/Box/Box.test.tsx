import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import 'jest-styled-components';

import Box, { Flex, Grid } from './Box';

describe('<Grid />', () => {
  it('should render a Grid content with two Boxes', () => {
    const { container } = render(
      <Grid>
        <Box>Box</Box>
        <Box>Box</Box>
      </Grid>
    );

    expect(container).toMatchSnapshot();
  });
});

describe('<Flex />', () => {
  it('should render a Flex content with two Boxes', () => {
    const { container } = render(
      <Flex>
        <Box>Box</Box>
        <Box>Box</Box>
      </Flex>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render a Flex with display inline-flex', () => {
    const { container } = render(
      <Flex inline>
        <Box>Box</Box>
        <Box>Box</Box>
      </Flex>
    );

    expect(container).toMatchSnapshot();
  });
});

describe('<Box />', () => {
  it('should render Box', () => {
    const { container } = render(<Box>Box</Box>);

    expect(container).toMatchSnapshot();
  });
});
