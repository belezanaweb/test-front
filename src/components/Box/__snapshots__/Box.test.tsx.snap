// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render Box 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0"
  >
    Box
  </div>
</div>
`;

exports[`<Flex /> should render a Flex content with two Boxes 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0"
    direction="row"
  >
    <div
      class="c1"
    >
      Box
    </div>
    <div
      class="c1"
    >
      Box
    </div>
  </div>
</div>
`;

exports[`<Flex /> should render a Flex with display inline-flex 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0"
    direction="row"
  >
    <div
      class="c1"
    >
      Box
    </div>
    <div
      class="c1"
    >
      Box
    </div>
  </div>
</div>
`;

exports[`<Grid /> should render a Grid content with two Boxes 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Box
    </div>
    <div
      class="c1"
    >
      Box
    </div>
  </div>
</div>
`;
